src/rez/package_order.py

```diff
@@ -472,6 +629,58 @@ def singleton(cls):
         """Filter list as configured by rezconfig.package_filter."""
         return cls.from_pod(config.package_orderers)
 
+    @staticmethod
+    def _to_orderer(orderer: Union[dict, PackageOrder]) -> PackageOrder:
+        if not isinstance(orderer, PackageOrder):
```

Comment:
```suggestion
        if isinstance(orderer, dict):
```

Reply:
I think that ensuring that is a `PackageOrder` is going to fail faster than just converting dicts to `PackageOrder`.  Under normal circumstances, a dict would be passed in, but we care a lot more about it being a PackageOrder not a not-dict.  It also means that if there were some change in the future to the type of data we can pass in here, we can accomodate that logic in the `from_pod()`, rather than having to deal with it here too.

Reply:
With a type hint, you are protected against future changes to types accepted for `orderer`. On top of that `from_pod` across the code base only accept dictionaries AFAIK and I doubt this will change in the future.

Reply:
That's fine.  What's the advantage to comparing it against a dict instead of `PackageOrder`?  It still seems less precise to me and less clear about what we're expecting to get from it at the end of the process.